src/foundry/client/pixi/layers/grid/layer.d.mts

```diff
@@ -37,187 +31,200 @@ declare global {
 
     /**
-     * Add highlighting for a specific grid position to a named highlight graphic
-     * @param name    - The name for the referenced highlight layer
-     * @param options - Options for the grid position that should be highlighted
+     * @deprecated since v12, will be removed in v14
+     * @remarks `"GridLayer#getTopLeft is deprecated. Use canvas.grid.getTopLeftPoint instead."`
      */
-    highlightPosition(name: string, options?: Parameters<BaseGrid["highlightGridPosition"]>[1]): false | void;
+    getTopLeft(x: number, y: number): PointArray;
 
     /**
-     * Test if a specific row and column position is a neighboring location to another row and column coordinate
-     * @param r0 - The original row position
-     * @param c0 - The original column position
-     * @param r1 - The candidate row position
-     * @param c1 - The candidate column position
+     * @deprecated since v12, will be removed in v14
+     * @remarks `"GridLayer#getCenter is deprecated. Use canvas.grid.getCenterPoint instead."`
      */
-    isNeighbor(r0: number, c0: number, r1: number, c1: number): boolean;
+    getCenter(x: number, y: number): PointArray;
+
+    /**
+     * @deprecated since v12, will be removed in v14
+     * @remarks `"GridLayer#getSnappedPosition is deprecated. Use canvas.grid.getCenterPoint instead."`
+     */
+    getSnappedPosition(
+      x: number,
+      y: number,
+      interval?: number,
+      options?: InexactPartial<{
+        /**
+         * The token
+         */
+        token: Token;
+      }>,
+    ): { x: number; y: number };
+
+    /**
+     * @deprecated since v12, will be removed in v14
+     * @remarks `"GridLayer#measureDistance is deprecated. "Use canvas.grid.measurePath instead for non-Euclidean measurements."`
+     */
+    measureDistance(
+      origin: {
+        x: number;
+        y: number;
+      },
+      target: {
+        x: number;
+        y: number;
+      },
+      options?: GridLayer.MeasureDistancesOptions,
+    ): number;
   }
 
   namespace GridLayer {
     interface LayerOptions extends CanvasLayer.LayerOptions {
       name: "grid";
     }
 
-    interface Segment {
-      ray: Ray;
-      label?: Ruler["labels"]["children"][number];
-    }
-
-    type DrawOptions = InexactPartial<{
+    interface HighlightPositionOptions {
       /**
-       * @defaultValue `null`
+       * The x-coordinate of the highlighted position
+       * @remarks Required if `canvas.grid !== CONST.GRID_TYPES.GRIDLESS`
        */
-      type?: foundry.CONST.GRID_TYPES | null;
+      x?: number | undefined | null;
 
       /**
-       * @defaultValue `null`
+       * The y-coordinate of the highlighted position
+       * @remarks Required if `canvas.grid !== CONST.GRID_TYPES.GRIDLESS`
        */
-      dimensions?: Canvas["dimensions"] | null;
+      y?: number | undefined | null;
 
-      color?: number | string | null;
+      /** The fill color of the highlight */
+      color?: PIXI.ColorSource | undefined;
```

Comment:
Is `null` an issue here? If so leaving a comment would be good.

Reply:
Yes, it's forwarded to a `PIXI.Graphics#beginFill` call which has a default value that `null` would override.